test/app-luatest/gh_7125-export-fiber_join_timeout_test.lua

```diff
@@ -0,0 +1,21 @@
+local t = require('luatest')
+local g = t.group()
+
+-- Check that fiber_join_timeout is accessible from ffi.
+g.test_ffi_fiber_join_timeout = function()
+    local t = require('luatest')
+    local ffi = require('ffi')
+    local fiber = require('fiber')
+    ffi.cdef([[
+        struct fiber;
+        struct fiber * fiber_self(void);
+        int fiber_join_timeout(struct fiber *f, double timeout);
+    ]])
+    local fiber_ptr = nil
+    local f = fiber.create(function()
+        fiber_ptr = ffi.C.fiber_self()
+        fiber.self():set_joinable(true)
+        fiber.sleep(0.01)
+    end)
+    t.assert_equals(ffi.C.fiber_join_timeout(fiber_ptr, 100500), 0)
```

Comment:
fiber_join_timeout yeilds. AFAIK it's a bit unsafe. https://github.com/tarantool/tarantool/commit/2711797be8c2e864db874a3f3a8434bc182d2360

Reply:
Oh, it seems that you are right. I'm sure btw that there's no chance that the bug you mentioned can fire here because tracing usually starts with hot path, but it should be a good practice to avoid ffi + yield.

Reply:
So I remove `fiber.sleep(0.01)` and now `fiber_join_timeout` has guarantee to finish without yield.